Code/GraphMol/Descriptors/Wrap/rdMolDescriptors.cpp

```diff
@@ -1048,4 +1134,80 @@ BOOST_PYTHON_MODULE(rdMolDescriptors) {
   python::def("CalcNumBridgeheadAtoms", numBridgeheadAtoms,
               (python::arg("mol"), python::arg("atoms") = python::object()),
               docString.c_str());
+
+  docString = "Property computation class stored in the property registry.\n"
+      "See rdkit.Chem.rdMolDescriptor.Properties.GetProperty and \n"
+      "rdkit.Chem.Descriptor.Properties.PropertyFunctor for creating new ones";
+  python::class_<RDKit::Descriptors::PropertyFunctor,
+                 RDKit::Descriptors::PropertyFunctor*,
+                 boost::shared_ptr<RDKit::Descriptors::PropertyFunctor>,
+                 boost::noncopyable>("PropertyFunctor", docString.c_str(), python::no_init)
+      .def("__call__", &RDKit::Descriptors::PropertyFunctor::operator(),
+           "Compute the property for the specified molecule")
+      .def("GetName", &RDKit::Descriptors::PropertyFunctor::getName,
+           "Return the name of the property to calculate")
+      .def("GetVersion", &RDKit::Descriptors::PropertyFunctor::getVersion,
+           "Return the version of the calculated property");
+
+  iterable_converter().from_python<std::vector<std::string> >();
+
+  docString = "Property computation and registry system.  To compute all registered properties:\n"
+      "mol = Chem.MolFromSmiles('c1ccccc1')\n"
+      "properties = rdMolDescriptors.Properties()\n"
+      "for name, value in zip(properties.GetPropertyNames(), properties.ComputeProperties(mol)):\n"
+      "  print(name, value)\n\n"
+      "To compute a subset\n"
+      "properties = rdMolDescriptors.Properties(['exactmw', 'lipinskiHBA'])\n"
+      "for name, value in zip(properties.GetPropertyNames(), properties.ComputeProperties(mol)):\n"
+      "  print(name, value)\n\n"
+      "";
+
+  python::class_<RDKit::Descriptors::Properties,
+                 RDKit::Descriptors::Properties*>("Properties", docString.c_str(), python::init<>() )
+      .def(python::init<const std::vector<std::string>&>())
+      .def("GetPropertyNames", &RDKit::Descriptors::Properties::getPropertyNames,
+           "Return the property names computed by this instance")
+      .def("ComputeProperties", &RDKit::Descriptors::Properties::computeProperties,
+           (python::arg("mol"), python::arg("annotateMol")=false),
+           "Return a list of computed properties, if annotateMol==True, annotate the molecule with "
+           "the computed properties.")
+      .def("annotateProperties", &RDKit::Descriptors::Properties::annotateProperties,
```

Comment:
The name exposed to Python should be `AnnotateProperties()`. I'll fix this post-merge.